app/controllers/saml_completion_controller.rb

```diff
@@ -8,9 +8,7 @@ class SamlCompletionController < ApplicationController
   def index
     if sp_session.present?
       request_url = URI(sp_session[:request_url])
-      path_year = request_url.path[-4..-1]
-      path_method = "api_saml_finalauthpost#{path_year}_url"
-      action_url = Rails.application.routes.url_helpers.send(path_method)
+      action_url = api_saml_finalauthpost_url(path_year: params[:path_year])
```

Comment:
so, ironically, this diff from #6894 that made me think we could use path parameters to let the router handle parsing.... this is currently the hardest part for me to refactor because we're storing the URL in the session and getting it back out, so we can't rely on the controller's `params` to have the correct value... not unless we add `?path_year=` to the `completions_url` ... so I'm currently thinking maybe we just close out this entire PR as not worth it... especially given all the spec changes I had to make to pass an additional param.

I think this is one of those cases where having a separate URL per year is really frustrating... but the cost to move away from that is not worth it at this time 😭 😎

Reply:
2023 update, now that I learned about `Rails.applicaiton.routes.recognize_path` in #8105, I think we can give this a shot!